samples/client/petstore/php-nextgen/OpenAPIClient-php/src/Api/FakeApi.php

```diff
@@ -3683,22 +3950,33 @@ public function testEndpointParametersRequest($number, $double, $pattern_without
-     * @param  string $enum_header_string Header parameter enum test (string) (optional, default to '-efg')
-     * @param  string[] $enum_query_string_array Query parameter enum test (string array) (optional)
-     * @param  string $enum_query_string Query parameter enum test (string) (optional, default to '-efg')
-     * @param  int $enum_query_integer Query parameter enum test (double) (optional)
-     * @param  float $enum_query_double Query parameter enum test (double) (optional)
-     * @param  \OpenAPI\Client\Model\EnumClass[] $enum_query_model_array enum_query_model_array (optional)
-     * @param  string[] $enum_form_string_array Form parameter enum test (string array) (optional, default to '$')
-     * @param  string $enum_form_string Form parameter enum test (string) (optional, default to '-efg')
+     * @param  string[]|null $enum_header_string_array Header parameter enum test (string array) (optional)
+     * @param  string|null $enum_header_string Header parameter enum test (string) (optional, default to '-efg')
+     * @param  string[]|null $enum_query_string_array Query parameter enum test (string array) (optional)
+     * @param  string|null $enum_query_string Query parameter enum test (string) (optional, default to '-efg')
+     * @param  int|null $enum_query_integer Query parameter enum test (double) (optional)
+     * @param  float|null $enum_query_double Query parameter enum test (double) (optional)
+     * @param  \OpenAPI\Client\Model\EnumClass[]|null $enum_query_model_array enum_query_model_array (optional)
+     * @param  string[]|null $enum_form_string_array Form parameter enum test (string array) (optional, default to '$')
+     * @param  string|null $enum_form_string Form parameter enum test (string) (optional, default to '-efg')
      * @param  string $contentType The value for the Content-Type header. Check self::contentTypes['testEnumParameters'] to see the possible values for this operation
      *
-     * @throws \OpenAPI\Client\ApiException on non-2xx response
-     * @throws \InvalidArgumentException
+     * @throws ApiException on non-2xx response
+     * @throws InvalidArgumentException
      * @return void
      */
-    public function testEnumParameters($enum_header_string_array = null, $enum_header_string = '-efg', $enum_query_string_array = null, $enum_query_string = '-efg', $enum_query_integer = null, $enum_query_double = null, $enum_query_model_array = null, $enum_form_string_array = '$', $enum_form_string = '-efg', string $contentType = self::contentTypes['testEnumParameters'][0])
+    public function testEnumParameters(
+        ?array $enum_header_string_array = null,
+        ?string $enum_header_string = '-efg',
+        ?array $enum_query_string_array = null,
+        ?string $enum_query_string = '-efg',
+        ?int $enum_query_integer = null,
+        ?float $enum_query_double = null,
+        ?array $enum_query_model_array = null,
+        ?array $enum_form_string_array = '$',
```

Comment:
This is obviously incorrect. The source in the api spec for this property says the following:
```yaml
properties:
  enum_form_string_array:
    description: Form parameter enum test (string array)
    type: array
    items:
      type: string
      default: $
      enum:
        - '>'
        - $
```

I think the default for this parameter should either be `['$']` or null.
Overriding the toDefaultValue ([see](https://github.com/OpenAPITools/openapi-generator/pull/16572/files#diff-759370b2185826fd9ae136c3f80da20d26ca5dba5f3cc52c6d0c3c80e1dc3fa7R192-R201)) also doesn't solve this problem.

If anyone knows how to fix this, please let me know.

Reply:
I'll try to take a look this weekend.

Reply:
I'll file another PR on top of this one to fix the issue.